cloud/aws/cloudfront/mode/discovery.pm

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright 2022 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package cloud::aws::cloudfront::mode::discovery;
+
+use base qw(centreon::plugins::mode);
+
+use strict;
+use warnings;
+use JSON::XS;
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+    
+    $options{options}->add_options(arguments => {
+        "prettify"  => { name => 'prettify' },
+    });
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::init(%options);
+}
+
+sub run {
+    my ($self, %options) = @_;
+    my @disco_data;
+    my $disco_stats;
+
+    $disco_stats->{start_time} = time();
+   
+    my $instances = $options{custom}->discovery(
+        service => 'cloudfront',
+        command => 'list-distributions'
+    );
+
+    foreach my $cft_instance (@{$instances->{DistributionList}->{Items}}) {   
+        next if (!defined($cft_instance->{Id}));
+        my %cft;    
+        $cft{id} = $cft_instance->{Id};
+        $cft{status} = $cft_instance->{Status};
+        $cft{domain_name} = $cft_instance->{DomainName};
+        $cft{aliases} = $cft_instance->{Aliases}->{Items};
```

Comment:
Aliases -> items can be not present (need to protect). And aliases is in hash table: DistributionConfig